tests/functional/z_cluster/test_enable_multus_existing_cluster.py

```diff
@@ -0,0 +1,272 @@
+"""
+Test module for enabling Multus networking on an existing OCS cluster.
+"""
+
+import logging
+import json
+import tempfile
+
+from ocs_ci.framework.testlib import polarion_id
+from ocs_ci.utility.reporting import get_polarion_id
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier2,
+    skipif_managed_service,
+    skipif_external_mode,
+    brown_squad,
+)
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.storage_cluster import (
+    get_storage_cluster,
+    verify_multus_network,
+    verify_storage_cluster,
+)
+from ocs_ci.ocs.resources.pod import get_osd_pods, wait_for_storage_pods
+from ocs_ci.utility import templating
+from ocs_ci.utility.utils import run_cmd, TimeoutSampler
+
+
+logger = logging.getLogger(__name__)
+
+
+@brown_squad
+@skipif_managed_service
+@skipif_external_mode
+@tier2
+@polarion_id(get_polarion_id())
+class TestEnableMultusExistingCluster(ManageTest):
+    """
+    Test class for enabling Multus on an existing cluster.
+    """
+
+    @tier2
+    def test_enable_multus_on_existing_cluster(self, pvc_factory, pod_factory):
+        """
+        Test enabling Multus networking on an existing OCS cluster.
+
+        Steps:
+        1. Verify StorageCluster is ready
+        2. Verify OSD pods are up and running
+        3. Install NMState operator and configure interfaces on compute nodes
+        4. Create NetworkAttachmentDefinition using existing templates
+        5. Update StorageCluster to use Multus networking
+        6. Wait for OSD pods to restart with Multus configuration
+        7. Verify Multus networking is configured on pods
+        8. Create RBD and CephFS PVCs and attach to pods
+        """
+        logger.info("Starting test to enable Multus on existing cluster")
+
+        logger.info("Step 1: Verifying StorageCluster is ready")
+        verify_storage_cluster()
+        logger.info("StorageCluster is Ready")
+
+        logger.info("Step 2: Verifying OSD pods are running")
+        wait_for_storage_pods()
+        osd_pods = get_osd_pods()
+        logger.info(f"All {len(osd_pods)} OSD pods are running")
+
+        sc = get_storage_cluster()
+
+        logger.info("Interface added to compute VMs successfully")
```

Comment:
I don't see in the test any adding of an interface. If we follow Oded suggestion and going only with Vsphere then you should use this function
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/vsphere.py#L1850